test/integration/exchangeIssuanceZeroExIntegration.spec.ts

```diff
@@ -0,0 +1,420 @@
+import "module-alias/register";
+import { Address, Account } from "@utils/types";
+import { ADDRESS_ZERO, MAX_UINT_256 } from "@utils/constants";
+import { SetToken } from "@utils/contracts/setV2";
+import { ether, getAccounts, getSetFixture, getWaffleExpect } from "@utils/index";
+import DeployHelper from "@utils/deploys";
+import { UnitsUtils } from "@utils/common/unitsUtils";
+import { SetFixture } from "@utils/fixtures";
+import { BigNumber, ContractTransaction } from "ethers";
+import { ExchangeIssuanceZeroEx, StandardTokenMock, WETH9 } from "@utils/contracts/index";
+import axios from "axios";
+import qs from "qs";
+import hre, { ethers } from "hardhat";
+
+const expect = getWaffleExpect();
+
+type SetTokenScenario = {
+  setToken: Address;
+  controller: Address;
+  issuanceModuleAddress: Address;
+  isDebtIssuanceModule: boolean;
+};
+
+type TokenName = "SimpleToken" | "DPI";
+
+function logVerbose(...args: any[]) {
+  if (process.env.VERBOSE) console.log(...args);
+}
+
+if (process.env.INTEGRATIONTEST) {
+  describe("ExchangeIssuanceZeroEx - Integration Test", async () => {
+    let owner: Account;
+    let user: Account;
+
+    let setV2Setup: SetFixture;
+    let deployer: DeployHelper;
+
+    // Contract Addresses
+    let wethAddress: Address;
+    let wbtcAddress: Address;
+    let daiAddress: Address;
+    let dpiAddress: Address;
+    let zeroExProxyAddress: Address;
+    let controllerAddress: Address;
+    let isDebtIssuanceModule: boolean;
+    let issuanceModuleAddress: Address;
+
+    // Contract Instances
+    let wbtc: StandardTokenMock;
+    let dai: StandardTokenMock;
+    let weth: WETH9;
+    let exchangeIssuanceZeroEx: ExchangeIssuanceZeroEx;
+    let simpleSetToken: SetToken;
+    let setToken: SetToken;
+
+    let daiUnits: BigNumber;
+    let wbtcUnits: BigNumber;
+
+    const setTokenScenarios: Partial<Record<TokenName, SetTokenScenario>> = {};
+    // Test Parameterization
+    const SET_TOKEN_NAMES: TokenName[] = ["DPI", "SimpleToken"];
+    const SET_TOKEN_AMOUNTS: Record<TokenName, number[]> = {
+      SimpleToken: [1],
+      // DPI issuance of 5000 failed due to underbuying Badger in a multihop
+      // TODO: Investigate
+      DPI: [3000, 1],
```

Comment:
Good test!

Reply:
Thanks, just noticed that the TODO comment above this line is outdated as well.

Reply:
Removed the todo comment